### PR TITLE
fix: extension nls resource initialization error

### LINKED ISSUE #3105: [BUG] 多语言模块有缺陷
我使用 registerLocalizationBundle 手动注册多语言文本，将 中文插件包中的多语言信息注册到 opensumi 中，以期望解决偶现的多语言失败的问题，但是现在的现象是部分 翻译成功，部分翻译失败。

目前针对的是 git 插件的中文语言包

**环境信息（Environment）**

- OS: windows11
- Browser: electron
- OpenSumi Version: 2.24.5

### PR DESCRIPTION
**Pull request by erha19**
### Types

- [x] 🐛 Bug Fixes

### <samp>🤖 Generated by Copilot at fadd4b5</samp>

*  Improve the performance and consistency of the extension service initialization by clearing the activated view extension map before the contributes services and executing them in parallel ([link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-3d705c2bf9f71760b12685e3d9e61fc6339b3886d9ace8c550014b9b2547b806L655-R657))
*  Simplify and optimize the logic of registering the language translations for each extension by using the spread operator, the map method, and a private storage path field ([link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-4eb5ad799a635b0ddb137686733cbe25351b3b2ef1794b26e8e19b5263feb66bL12-R12), [link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-4eb5ad799a635b0ddb137686733cbe25351b3b2ef1794b26e8e19b5263feb66bR66-R67), [link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-4eb5ad799a635b0ddb137686733cbe25351b3b2ef1794b26e8e19b5263feb66bL90-R116))
*  Fix a bug in the `registerLanguage` method of the localization contribution point class, where the language bundle file was not converted to a string before parsing it as JSON ([link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-4eb5ad799a635b0ddb137686733cbe25351b3b2ef1794b26e8e19b5263feb66bL121-R127))
*  Add a `Throttler` instance to the `ExtensionContributesService` class to manage the execution of the contribute points according to the lifecycle phases ([link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-96be7d1ed582a90bb723f1e64a8f7c8c690c5fecc8a7c628135080e3d9819d65R24), [link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-96be7d1ed582a90bb723f1e64a8f7c8c690c5fecc8a7c628135080e3d9819d65R331-R332), [link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-96be7d1ed582a90bb723f1e64a8f7c8c690c5fecc8a7c628135080e3d9819d65L402-R415))
*  Add an optional parameter `from` to the `contribute` method of the `VSCodeContributePoint` class, to indicate the source of the contribution request and allow the subclasses to handle different scenarios ([link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-96be7d1ed582a90bb723f1e64a8f7c8c690c5fecc8a7c628135080e3d9819d65L279-R280), [link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-96be7d1ed582a90bb723f1e64a8f7c8c690c5fecc8a7c628135080e3d9819d65L372-R374))
*  Rename and refactor the `runContributes` method of the `ExtensionContributesService` class, to initiate and execute the contribution requests for a given lifecycle phase ([link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-96be7d1ed582a90bb723f1e64a8f7c8c690c5fecc8a7c628135080e3d9819d65L392-R395))
*  Wait for the theme change event to fire once before restoring the terminal views in the `TerminalLifeCycleContribution` class, to ensure the correct theme and colors are applied ([link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-b97d2a705f01f4623b3a3e1d017d16bd62f790af066f0fc71fae200885e6ea72L2-R4), [link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-b97d2a705f01f4623b3a3e1d017d16bd62f790af066f0fc71fae200885e6ea72R25-R27), [link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-b97d2a705f01f4623b3a3e1d017d16bd62f790af066f0fc71fae200885e6ea72L35-R43))
*  Make the `terminalProfileInternalService` and `variableResolver` fields of the `TerminalClient` class public, and move the initialization of the `xterm` field to a private async method ([link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-a8de8caed40e0946a999164ec3ab3b4c053a2abf0bd1ac6365ef89c39167b744L131-R134), [link](https://github.com/opensumi/core/pull/3133/files?diff=unified&w=0#diff-a8de8caed40e0946a999164ec3ab3b4c053a2abf0bd1ac6365ef89c39167b744L163-R166))

close #3105.

### <samp>🤖 Generated by Copilot at fadd4b5</samp>

This pull request improves and enhances various modules related to extension and terminal contributions. It fixes a bug in localization, simplifies the registration logic, handles different sources and phases of contribution requests, handles theme changes in terminal views, and improves the performance and consistency of extension service initialization. It also makes some minor changes to the visibility and initialization of some fields.

### CODE REVIEW

File: packages/extension/src/common/extension.ts [hunk 1]
```diff
@@ -388,27 +390,40 @@ export abstract class ExtensionContributesService extends WithEventBus {
...
-
-      await this.runContributesByPhase(phase);
-    };
-
-    this.addDispose(
-      this.lifecycleService.onDidLifeCyclePhaseChange((newPhase) => {
-        runContributes(newPhase);
-      }),
-    );
-    await runContributes();
```
Comment (erha19) [inline packages/extension/src/common/extension.ts:411]: 核心修改位于这里，原有的逻辑至做了根据 LifeCyclePhase 执行的逻辑，但没有保证时序，故而导致当有 NLS 插件时，出现初始化和框架后续视图（包含 Menu、WelcomePage 等）渲染上的竞速问题，这里的修改用队列保证了这块时序可靠性。

**Review by Ricbet — DISMISSED** [review]

**Review by bytemain — APPROVED** [review]

**Pull request merged**